data_hub_metrics_api/page_views_and_downloads_provider.py

```diff
@@ -151,20 +148,26 @@ def refresh_page_view_and_download_totals(self) -> None:
         total_rows = bq_result.total_rows
         LOGGER.info('Total rows from BigQuery: %d', total_rows)
 
-        for row in tqdm(bq_result, total=total_rows, desc="Loading Redis"):
-            self.redis_client.set(
-                f'article:{row['article_id']}:page_views',
-                row['page_view_count']  # type: ignore[arg-type]
-            )
-            self.redis_client.set(
-                f'article:{row['article_id']}:downloads',
-                row['download_count']  # type: ignore[arg-type]
-            )
+        with self.redis_client.pipeline() as pipe:
+            for i, row in enumerate(iter_with_progress(bq_result, total_rows, 'Loading Redis'), 1):
+                pipe.set(
+                    f'article:{row['article_id']}:page_views',
+                    row['page_view_count']  # type: ignore[arg-type]
+                )
+                pipe.set(
+                    f'article:{row['article_id']}:downloads',
+                    row['download_count']  # type: ignore[arg-type]
+                )
+                if i % batch_size == 0:
```

Comment:
minor: I believe this makes the first batch contain `1` item instead of `1000`. Probably doesn't make a real world difference.
(In some other repo we had a utility function to split things into batches)